stock_indicators/indicators/common/candles.py

```diff
@@ -1,21 +1,105 @@
-
 from decimal import Decimal
 from typing import Optional
 
+from typing_extensions import Self
+
 from stock_indicators._cstypes import Decimal as CsDecimal
 from stock_indicators._cstypes import to_pydecimal
-from stock_indicators.indicators.common.results import ResultBase
+from stock_indicators.indicators.common.enums import Signal
+from stock_indicators.indicators.common.quote import Quote
+from stock_indicators.indicators.common.results import IndicatorResults, ResultBase
+
+
+class CondenseMixin:
+    """Mixin for condense()."""
+    @IndicatorResults._verify_data
+    def condense(self) -> Self:
+        """
+        Remove non-essential results so it only returns meaningful data records.
+        """
+        return self.__class__(filter(lambda x: x.signal != Signal.NONE, self), self._wrapper_class)
```

Comment:
1. `condense()` does not call DLL. Instead, it removes meaningless records on Python-side.
2. If there is better descriptive docstring for `condense()`, plz update.

Reply:
I think this is okay.  See [utilities](https://daveskender.github.io/Stock.Indicators/utilities/#condense)

Reply:
Yeah, I copied from that haha.